### DISCUSSION TITLE
Smooth Scroll when scrolling with Scroll Bar Thumbs

### DISCUSSION THREAD

**Discussion by blitzve0** [idiotWu/smooth-scrollbar#394] [Ideas]
Oh I'm so sorry the title is wrong it was a wrong autofill
What I initially wrote was "Having Smooth Scroll while scrolling with the scrollbar thumb", as currently smooth Scroll only happens when we use mouse wheel or arrow keys on keyboard to scroll. If we scroll with the scrollbar at right side that scroll is plain not smooth

Old : Due to this issue i am on the verge of shifting to Locomotive scroll as when we scroll using the scrollbar the smooth scroll doesn't work

**Comment by dolphin-wood** [comment]
I think this is mainly because that we don't emit `scroll` events. Maybe we can fire custom events for compatibility reasons 🤔  (not sure if it will work).

**Reply by blitzve0** [reply]
Oh I'm so sorry the title is wrong it was a wrong autofill
What I initially wrote was "Having Smooth Scroll while scrolling with the scrollbar thumb", as currently smooth Scroll only happens when we use mouse wheel or arrow keys on keyboard to scroll. If we scroll with the scrollbar at right side that scroll is plain not smooth

**Reply by dolphin-wood** [reply]
It's quite easy to smoothen the scrolling when dragging scrollbar thumbs, but it will make the scrolling somehow laggy due to the easing.

**Reply by dolphin-wood** [reply]
@blitzve0 you can change [these lines](https://github.com/idiotWu/smooth-scrollbar/blob/7d4057dee4f83672cda89544b658f06fe1295470/src/events/mouse.ts#L140-L143) to

```js
scrollbar.setMomentum(
  0,
  calcOffset(startTrackDirection, offsetOnTrack) - offset.y,
);
```

to see how it works in your local fork. The scrolling will be less responsive with a small damping value (e.g. `< 0.05`).

**Reply by dolphin-wood** [reply]
Hmm....maybe it's not that bad as I've imagined. Now that everything has been smoothened, why not this one? I'll add this to the next release now 😆

**Reply by blitzve0** [reply]
Thanks Man, also I think modern web browsers can take any thing without slowing down

**Comment by dolphin-wood** [comment]
Shipped as [8.7.0](https://github.com/idiotWu/smooth-scrollbar/releases/tag/v8.7.0) 🎉

**Reply by blitzve0** [reply]
Damn You literally solved and shipped it in a day, 😍😍😍

Update : Just implemented the latest version, life saved, now i don't have to discard all my projects which used thuis plugin, its smooth as hell

**Comment by Discreater** [comment]
How to disable smooth when draggin the thumb? The thumb falls behind my mouse by a lot.
